torch/_inductor/codegen/cpp.py

```diff
@@ -357,6 +359,198 @@ def stride_at_vec_range(index: sympy.Expr, var: sympy.Symbol, vec_length: int):
+    ):
+        # This function ensures that the same tiling split is applied at each loop level within the outer loop fusion depth.
+        # In the fusion stage, we only examine nodes with same vars and reduce.
+        # However, for nodes with same vars and reduce, the loops may still have different tile splits.
+        # For example (test_expr_vec_non_contiguous in test_cpu_repro.py):
+        #   * buf0 tiling along the 2nd loop level, buf1 tiling along the 3rd loop level.
+        # If the check failed, we should fall back to standard loop codegen.
+        def _inner(
+            _left_loop_level: LoopLevel,
+            _right_loop_level: LoopLevel,
+            _loop_fusion_depth: int,
+        ) -> bool:
+            # Check if same loop level attr
+            _outer_loops_attr_compare_list = [
+                "var",
+                "size",
+                "offset",
+                "steps",
+            ]
+            if not (
+                all(
+                    getattr(_left_loop_level, _attr_compare)
+                    == getattr(_right_loop_level, _attr_compare)
+                    for _attr_compare in _outer_loops_attr_compare_list
+                )
+            ):
+                return False
+
+            assert _loop_fusion_depth >= 1
+            if (_loop_fusion_depth := _loop_fusion_depth - 1) > 0:
+                # If the next loop level is expected to undergo outer loop fusion,
+                # there should be no kernel present at the current loop level.
+                assert (
+                    _left_loop_level.kernel is None and _right_loop_level.kernel is None
+                )
+                # Check next loop level attr
+                if len(_left_loop_level.inner) != len(_right_loop_level.inner) or any(
+                    not _inner(left_inner, right_inner, _loop_fusion_depth)
+                    for left_inner, right_inner in zip(
+                        _left_loop_level.inner, _right_loop_level.inner
+                    )
+                ):
+                    return False
+            return True
+
+        def _get_parallel_depth(_cpp_proxy_kernel):
+            return _cpp_proxy_kernel.decide_parallel_depth(
+                _cpp_proxy_kernel.call_ranges[
+                    : _cpp_proxy_kernel.loop_nest.max_parallel_depth()
+                ],
+                parallel_num_threads(),
+            )
+
+        for idx in range(len(_cpp_kernel_proxy_list) - 1):
+            _left_loop_nest = _cpp_kernel_proxy_list[idx].loop_nest
+            _right_loop_nest = _cpp_kernel_proxy_list[idx + 1].loop_nest
+            if _get_parallel_depth(_cpp_kernel_proxy_list[idx]) != _get_parallel_depth(
+                _cpp_kernel_proxy_list[idx + 1]
+            ):
+                # Ensure same parallel depth.
+                # Refer to poolformer_m36 in timm_models:
+                # Node1 has loop collapsed(2), Node2 can't be loop collapsed due to it's reduction.
```

Comment:
Thanks for the comment. This check has been removed. After we refactor to use the `OuterLoopFusedKernel`, this issue has gone. I think that's because [`max_parallel_depth`](https://github.com/pytorch/pytorch/blob/2b4173e0def678b04efea46a85909bb86f7c3fd6/torch/_inductor/codegen/cpp.py#L4087-L4090) only counts the loop level which has inners. After the refactor, it's same as the outer loop fusion depth.